python/code/classificationmethods/ManifoldDiscrimantAnalysisParafacTucker.py

```diff
@@ -0,0 +1,193 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from scipy.stats import ortho_group  
+from pymanopt import Problem
+from pymanopt.manifolds import Product, Stiefel
+import tensorflow as tf
+from pymanopt.solvers import ConjugateGradient
+from sklearn.pipeline import Pipeline
+
+
+class ManifoldDiscrimantAnalysis(ABC, Pipeline):
+    def __init__(self, classes=None, opts=None, usestoppingcrit=True, maxits=1000,
+                 store={'Rw': None, 'Rb': None, 'QtRb': None, 'QtRw':  None, 'QtBQ': None, 'QtWQ': None, 'QtWQinvQtBQ': None},
+                 optmeth='ManOpt'):
+        self.classes = classes
+        self.opts = opts
+        self.usestoppingcrit = usestoppingcrit
+        self.maxits = maxits
+        self.store = store
+        self.optmeth = optmeth
+        if optmeth not in ['bo13', 'wen12', 'ManOpt']:
+            self.optmeth = 'ManOpt'
+            print("Chosen optimization method", optmeth, "has not been implemented")
+        self.Fdifftol = Fdifftol = 10**(-10)
+        self.Udifftol = Udifftol = 10**(-12)
+        self.MyCost = None
+        self.rotations = None
+
+
+    def set_tolerances(self, Fdifftol=10**(-10), Udifftol=10**(-12)):
+        self.Fdifftol = Fdifftol
+        self.Udifftol = Udifftol
+        
+    @abstractmethod 
+    def object_matrix_data(self, classmeandiffs, observationdiffs, nis, K1, K2, ): #This is a virtual function that any inheriting subclass should realize
```

Comment:
This line seems quite long, can we have the comment above the function definition?